Kconfig

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2017 Linaro Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+mainmenu "MCUBOOT configuration"
+
+config ZEPHYR_BASE
+	string
+	option env="ZEPHYR_BASE"
+
+config APPLICATION_BASE
+	string
+	option env="PROJECT_BASE"
+
+source "$ZEPHYR_BASE/Kconfig.zephyr"
+
+menuconfig MCUBOOT_SERIAL
+	bool
+	prompt "MCUBOOT serial recovery"
+	default n
+	select REBOOT
+	select UART_INTERRUPT_DRIVEN
+	select SERIAL
+	select NEWLIB_LIBC
+	help
+	  Enable serial recovery feature in mcuboot.
+
+config BOOT_MAX_LINE_INPUT_LEN
+	int "cmd line max length" 
+	default 512
+	depends on MCUBOOT_SERIAL
+	help
+	  Maximum length of commands transported over the serial protocol
+
+config BOOT_SERIAL_DETECT_PORT
+	string
+	prompt "GPIO Port for triger serial recovery mode"
+	default CONFIG_GPIO_NRF5_P0_DEV_NAME
```

Comment:
Should not this config be BSP specific somehow?

Reply:
It is configuration for the zephyr application which in this case is mcuboot. It shouldn't be a part of zephyr-project sources as this peculiar configuration.

Reply:
Or maybe there should not be a default at all here? This needs to be properly configured when building MCUboot with serial recovery. Instead I would fail early if this is an empty string?

Reply:
I would convert it to:
```default CONFIG_GPIO_NRF5_P0_DEV_NAME if SOC_FAMILY_NRF5```
?